nncf/onnx/statistics/collectors.py

```diff
@@ -76,17 +74,12 @@ def masked_mean(
     ) -> NNCFTensor:
         if mask is None:
             return cls.mean(x, axis=axis, keepdims=keepdims)
+        comp_dtype, out_dtype = _get_computing_dtype(x.tensor.dtype)
         masked_x = np.ma.array(x.tensor, mask=mask.tensor)
-        if x.tensor.dtype in [np.float32, np.float16]:
-            result = np.ma.mean(masked_x, axis=axis, keepdims=keepdims, dtype=np.float64)
-            if isinstance(result, np.ma.MaskedArray):
-                result = result.data
-            return ONNXNNCFTensor(result.astype(dtype=x.tensor.dtype))
-        else:
-            result = np.ma.mean(masked_x, axis=axis, keepdims=keepdims)
-            if isinstance(result, np.ma.MaskedArray):
-                result = result.data
-            return ONNXNNCFTensor(result)
+        result = np.ma.mean(masked_x, axis=axis, keepdims=keepdims, dtype=comp_dtype)
+        if isinstance(result, np.ma.MaskedArray):
+            result = result.data
```

Comment:
Don't forget to specify return type in this case

Reply:
This is done below
`return ONNXNNCFTensor(result.astype(dtype=out_dtype, copy=False))`